franklin/github/views.py

```diff
@@ -190,3 +196,66 @@ def deploy_hook(request):
 def get_user_repos(request):
     if request.method == 'GET':
         return Response(request.user.details.get_user_repos())
+
+def get_access_token(request):
+    """
+    Tries to get the access token from an OAuth Provider
+    :param request:
+    :param backend:
+    :return:
+    """
+    access_token_url = 'https://github.com/login/oauth/access_token'
+    secret = github_secret
+
+    params = {
+        'client_id': request.data.get('clientId'),
+        'redirect_uri': request.data.get('redirectUri'),
+        'client_secret': secret,
+        'code': request.data.get('code'),
+        'accept': 'json'
+    }
+
+    # Exchange authorization code for access token.
+    r = requests.get(access_token_url, params=params)
```

Comment:
for this part, check out core > helpers. I created GET/POST helper methods there that do the validation for you. Thus, you can trust the response coming back.